Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -1012,6 +1035,90 @@ public final class HTTPClient: Sendable {
         }
+
+        public init(
+            tlsConfiguration: TLSConfiguration? = nil,
+            redirectConfiguration: RedirectConfiguration? = nil,
+            timeout: Timeout = Timeout(),
+            connectionPool: ConnectionPool = ConnectionPool(),
+            proxy: Proxy? = nil,
+            ignoreUncleanSSLShutdown: Bool = false,
+            decompression: Decompression = .disabled,
+            http1_1ConnectionDebugInitializer: (@Sendable (Channel) -> EventLoopFuture<Void>)? = nil,
+            http2ConnectionDebugInitializer: (@Sendable (Channel) -> EventLoopFuture<Void>)? = nil,
+            http2StreamChannelDebugInitializer: (@Sendable (Channel) -> EventLoopFuture<Void>)? = nil,
+            tracing: TracingConfiguration = .init()
+        ) {
+            self.init(
+                tlsConfiguration: tlsConfiguration,
+                redirectConfiguration: redirectConfiguration,
+                timeout: timeout,
+                connectionPool: connectionPool,
+                proxy: proxy,
+                ignoreUncleanSSLShutdown: ignoreUncleanSSLShutdown,
+                decompression: decompression
+            )
+            self.http1_1ConnectionDebugInitializer = http1_1ConnectionDebugInitializer
+            self.http2ConnectionDebugInitializer = http2ConnectionDebugInitializer
+            self.http2StreamChannelDebugInitializer = http2StreamChannelDebugInitializer
+            self.tracing = tracing
+        }
+    }
+
+    public struct TracingConfiguration: Sendable {
+
+        @usableFromInline
+        var _tracer: Optional<any Sendable>  // erasure trick so we don't have to make Configuration @available
+
+        /// Tracer that should be used by the HTTPClient.
+        ///
+        /// This is selected at configuration creation time, and if no tracer is passed explicitly,
+        /// (including `nil` in order to disable traces), the default global bootstrapped tracer will
+        /// be stored in this property, and used for all subsequent requests made by this client.
+        @inlinable
+        @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+        public var tracer: (any Tracer)? {
+            get {
+                guard let _tracer else {
+                    return nil
+                }
+                return _tracer as! (any Tracer)?
+            }
+            set {
+                self._tracer = newValue
+            }
+        }
+
+        /// Configuration for tracing attributes set by the HTTPClient.
+        public var attributeKeys: AttributeKeys
+
+        public init() {
+            self._tracer = nil
+            self.attributeKeys = .init()
+        }
+
+        @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+        public init(
+            tracer: (any Tracer)? = InstrumentationSystem.tracer,
+            attributeKeys: AttributeKeys = .init()
+        ) {
+            self._tracer = tracer
+            self.attributeKeys = attributeKeys
+        }
+
```

Comment:
Do we need a callback to allow overwriting the span name?

Reply:
Maybe, I'd love to keep all the extra customization to future PRs though